src/Stratis.Bitcoin.Features.BlockStore.Tests/BlockStoreTests.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+using Moq;
+using NBitcoin.DataEncoders;
+
+namespace Stratis.Bitcoin.Features.BlockStore.Tests
+{
+    public class BlockStoreTests
+    {
+        private BlockStore blockStore;
+        private BlockRepositoryMock repository;
+        private ConcurrentChain chain;
+        private IChainState chainState;
+        private NodeLifetime nodeLifetime;
+
+        private ChainedHeader consensusTip;
+
+        public BlockStoreTests()
+        {
+            this.repository = new BlockRepositoryMock();
+
+            this.chainState = new Mock<IChainState>().SetupProperty(x => x.ConsensusTip, this.consensusTip).Object;
+
+            this.chain = this.CreateChain(10);
+            this.consensusTip = this.chain.Tip;
+            this.nodeLifetime = new NodeLifetime();
+
+            this.blockStore = new BlockStore(this.repository, this.chain, this.chainState, new StoreSettings(), this.nodeLifetime, new LoggerFactory());
+        }
+
+        private ConcurrentChain CreateChain(int blocksCount)
+        {
+            var chain = new ConcurrentChain(Network.StratisMain);
+
+            for (int i = 0; i < blocksCount; i++)
+            {
+                var header = new BlockHeader()
+                {
+                    Nonce = RandomUtils.GetUInt32(),
+                    HashPrevBlock = chain.Tip.HashBlock,
+                    Bits = Target.Difficulty1
+                };
+                
+                var chainedHeader = new ChainedHeader(header, header.GetHash(), chain.Tip);
+
+                chain.SetTip(chainedHeader);
+            }
+
+            return chain;
+        }
+
+        private async Task WaitUntilQueueIsEmpty()
+        {
+            while (this.blockStore.BlocksQueueCount != 0)
+                await Task.Delay(100).ConfigureAwait(false);
+            
+            // For very slow environments.
+            await Task.Delay(100).ConfigureAwait(false);
+        }
+
+        /// <summary>Checks that block store tip initializes from the hash of the last saved block.</summary>
+        [Fact]
+        public async Task BlockStoreInitializesTipAtHashBlockAsync()
```

Comment:
This test name is almost the same as the summary, I would suggest not duplicating this information. The test name could be extended slightly to include (of the last saved block - if that info is important) and the summary deleted, thus saving future readers of the code the effort of having to read the summary and test name and the long term overhead of keeping both in sync.

Reply:
Fixed

Reply:
@noescape00 reminder comments on test methods are not really required.